app/src/main/java/org/bepass/oblivion/OblivionVpnService.java

```diff
@@ -447,6 +449,15 @@ private void configure() {
                     .addDisallowedApplication(getPackageName())
                     .addRoute("0.0.0.0", 0)
                     .addRoute("::", 0);
+            fileManager.getStringSet("splitTunnelApps", new HashSet<>());
+            SplitTunnelMode splitTunnelMode = SplitTunnelMode.getSplitTunnelMode(fileManager);
+            if (splitTunnelMode != SplitTunnelMode.DISABLED) {
+                for (String packageName : getSplitTunnelApps(fileManager)) {
+                    if (splitTunnelMode == SplitTunnelMode.BLACKLIST)
+                        builder.addDisallowedApplication(packageName);
+                }
```

Comment:
You are right. It's like this because we had two modes at first (BLACKLIST and WHITELIST) but then I realized the whitelist mode required more work from the Go core lib side so I just removed the WHITELIST branch from the conditions. I will simplify the leftovers.